### ISSUE TITLE
ECONNRESET while scanning library

### ISSUE THREAD

**Issue by blyndfreddy** [giwty/nsp-update#74]
Program starts, checks to download titles.json and versions.json from tinfoil, and begins scanning Library, and generates Javascript ECONNRESET error and closes. Error first seen 24th July 2021. Tried on Windows 10 LTSC and Windows 10 Pro 21H1, with Wifi, and Wifi + VPN and no net connection at all. Really liked SLM but been a while since update. Is project dead?
[slm.log](https://github.com/giwty/switch-library-manager/files/6873262/slm.log)

**Comment by giwty** [comment]
Just tested and it works on my end, also not seeing any errors on the slm.log.
Can you provide screenshot of the error ?
You can also try to open dev tools under the debug menu and see if there is any errors there.

Are you using a network drive ?

Project is not dead, i just have less time, and feel it's still in good condition :)

**Comment by blyndfreddy** [comment]
Yep, I looked at the log too, and it doesn't give much away. It looks as if the error happens as soon as it starts scanning the Library. Why it would generate a TCP error when it doesn't appear to touch the network is odd. I thought my VPN was causing an issue so I tried without it, and also without a connection at all. All situations generated the same error. The Library is on a removable USB Hard disk, but I've been using it like that for a while with no problems. 
![econnreset](https://user-images.githubusercontent.com/87919575/126915161-304bc32a-3ff8-4d9a-805f-1b1a2bd8b169.png)

**Comment by giwty** [comment]
The connection error is between the UI and the "server" - has nothing to do with outgoing connection.
If you have an existing installation, try to re-download and start fresh.
It's weird that nothing is captured in the slm.iog
Did you check the dev tools? looks at the console tab.

**Comment by blyndfreddy** [comment]
Ok I've attached some photos. Hope they help.
![Slm_Error](https://user-images.githubusercontent.com/87919575/126937058-c13076cf-82fc-499e-aad7-fc008e209c6e.jpg)
![Slm_DevTools](https://user-images.githubusercontent.com/87919575/126937076-b60deee7-3e1a-4fa5-80a5-39ba3b7b35d0.jpg)
![Slm_DevTools_Console](https://user-images.githubusercontent.com/87919575/126938263-f139c096-f5d5-41b2-86bb-944c99bad911.jpg)
![Slm_Console](https://user-images.githubusercontent.com/87919575/126938273-1edc6d46-5452-453e-bcc5-cccbec6e2e4e.jpg)

**Comment by blyndfreddy** [comment]
Just looking at the slm.log I had a feeling the Diablo 3 directory had problems. I removed it from the Library and the program ran properly. I removed the recent additions (Language packs as split files) and restored the directory. And all's well. Thanks in any case for looking at the problem. I guess this makes the case for an enhancement where non-conforming files are flagged. I love the program. SLM is a vital part of my Stash management. Best wishes in the future.
